### PR TITLE
api: update the generator with tarantool 3.2.0

### LINKED ISSUE #6: Update the generator
It looks like our generator is outdated. We need to update it.
```
$ TT_TAG=3.2.0 make
...
+ rm -rf tarantool
stringer: no values defined for type Error
generate.go:4: running "stringer": exit status 1
make: *** [Makefile:17: generate] Error 1
```

### PR DESCRIPTION
**Pull request by dmyger**
Fix parsing box/errcode.h constants.

Closes #6

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -3,163 +3,171 @@
...
 	ER_MEMORY_ISSUE Error = 2
-	// "Duplicate key exists in unique index \"%s\" in space \"%s\" with old tuple - %s and new tuple - %s"
+	// nolint:lll
+	// "Duplicate key exists in unique index \"%s\" in space \"%s\" with old tuple - %s and new tuple - %s" "index" STRING "space" STRING "" STRING "" STRING "old_tuple" TUPLE "new_tuple" TUPLE
 	ER_TUPLE_FOUND Error = 3
-	// "Tuple doesn't exist in index '%s' in space '%s'"
+	// Unused
 	ER_TUPLE_NOT_FOUND Error = 4
-	// "%s does not support %s"
+	// "%s does not support %s" "" STRING "" STRING
```
Comment (oleg-jukovec) [inline error.go]: Let's left only the error message here. The comments at now looks confusing and I don't see much sense in the rest of the parts.
```suggestion
	// "%s does not support %s"
```

File: error.go [hunk 2]
```diff
@@ -3,163 +3,171 @@
...
 const (
 	// "Unknown error"
 	ER_UNKNOWN Error = 0
-	// "Illegal parameters, %s"
+	// "%s" "" STRING
 	ER_ILLEGAL_PARAMS Error = 1
 	// "Failed to allocate %u bytes in %s for %s"
 	ER_MEMORY_ISSUE Error = 2
-	// "Duplicate key exists in unique index \"%s\" in space \"%s\" with old tuple - %s and new tuple - %s"
+	// nolint:lll
```
Comment (oleg-jukovec) [inline error.go]: The nolint will be removed then.

File: flag.go [hunk 3]
```diff
@@ -3,7 +3,7 @@
 package iproto
 
 // IPROTO flag constants, generated from
-// tarantool/src/box/iproto_constants.h
+// tarantool-master/src/box/iproto_constants.h
```
Comment (oleg-jukovec) [inline flag.go]: It looks like an unnecessary change when we will regenerate the file for a next tag.
```suggestion
// src/box/iproto_constants.h
```

File: generate.sh [hunk 4]
```diff
@@ -11,12 +11,12 @@ if [[ -z "${TT_REPO}" ]]; then
 	TT_REPO="https://github.com/tarantool/tarantool.git"
 fi
 
-TT_DIR=tarantool
+TT_DIR=tarantool-$TT_TAG
```
Comment (oleg-jukovec) [inline generate.sh]: Just to be with the same style.
```suggestion
TT_DIR=tarantool-${TT_TAG}
```

Comment (oleg-jukovec) [inline generate.sh]: I understand why this is done, but it is really unlikely that I will need different versions of tarantool when running the generator. So up to you, be free to ignore the suggestion.
```suggestion
TT_DIR=tarantool
```

File: generate.sh [hunk 5]
```diff
@@ -233,19 +233,22 @@ type Error int
...
-	awk '{
+# Extract body of define ERROR_CODES
+ERROR_CODES=`sed -n '/^#define ERROR_CODES(/,/This one should be last/{//!p;}' ${SRC_ERRORS}`
+
+echo "${ERROR_CODES}" | \
+	# Get line with ER_* description
+	sed -nE 's/^[^E]*(ER_.+)\s*\\\s*$/\1/p'  | \
+	# Remove trailing parenthesis or comment's ending
+	sed -E 's/(.*)\s*(:?\)|\*\/)\s*$/\1/' | \
+	awk '
```
Comment (oleg-jukovec) [inline generate.sh]: Please, comment each line. It is really hard to understand what happened here after a few month. Comments may help.
```suggestion
	sed -E 's/(.*)\s*(:?\)|\*\/)\s*$/\1/' | \
	# Foo
	awk '
```

**Review by oleg-jukovec — COMMENTED** [review]
Thank you for the patch! Looks great, but I have few non-critical comments.

**Comment by oleg-jukovec** [comment]
Please, add a changelog entry.

File: generate.sh (hunk 4 above)
Comment (dmyger) [reply generate.sh]: I would prefer to keep it. As it is useful for debugging, when you need keep both versions to compare them.

File: error.go (hunk 2 above)
Comment (dmyger) [reply error.go]: Not in this case. Even if we delete the rest of it, this comment will be over 100 characters anyway.

File: error.go (hunk 1 above)
Comment (dmyger) [reply error.go]: Ok, except that would require complicating the regexp.

File: error.go (hunk 2 above)
Comment (oleg-jukovec) [reply error.go]: Let's disable the `lll` linter. All our files are generated and there is no much sense in that.

It could be done here:
https://github.com/tarantool/go-iproto/blob/402077faa673942675215b6cae97c95693269574/.github/workflows/test.yml#L84

and here:

https://github.com/tarantool/go-iproto/blob/402077faa673942675215b6cae97c95693269574/.github/workflows/test.yml#L91

with `-D lll` I guess.

File: generate.sh (hunk 4 above)
Comment (oleg-jukovec) [reply generate.sh]: Ok than.

File: error.go (hunk 2 above)
Comment (dmyger) [reply error.go]: Just removed adding this `nolint`. Cause, generated files ignored by linter.

File: CHANGELOG.md [hunk 6]
```diff
@@ -11,6 +11,7 @@ Versioning](http://semver.org/spec/v2.0.0.html) except to the first release.
 ### Added
 
 ### Changed
+- Update parsing enum macro of `errcode.h` from Tarantool 3.2.0 (#6)
```
Comment (oleg-jukovec) [inline CHANGELOG.md]:
```suggestion
### Changed

- Update parsing enum macro of `errcode.h` from Tarantool 3.2.0 (#6).
```

**Review by oleg-jukovec — APPROVED** [review]
Thank you for the patch, just a one small fix required.

File: CHANGELOG.md (hunk 6 above)
Comment (dmyger) [reply CHANGELOG.md]: Done.

**Review by oleg-jukovec — APPROVED** [review]

**Review by DerekBum — APPROVED** [review]
